### PR TITLE
Apply small polish to cache code

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: docs/src/main/asciidoc/cache.adoc [hunk 1]
```diff
@@ -539,19 +539,22 @@ Another way to retrieve a `Cache` instance consists in injecting the `io.quarkus
...
 import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
 
 import io.quarkus.cache.Cache;
 import io.quarkus.cache.CacheManager;
 
 import java.util.Optional;
 
-@ApplicationScoped
+@Singleton
```
Comment (gwenneg) [inline docs/src/main/asciidoc/cache.adoc:550]: I'm curious. Is `@Singleton` the new recommended scope by default with Quarkus 3.0? The doc from 2.x still [recommends](https://quarkus.io/guides/cdi#applicationscoped-and-singleton-look-very-similar-which-one-should-i-choose-for-my-quarkus-application) to use `@ApplicationScoped`.

Comment (geoand) [reply docs/src/main/asciidoc/cache.adoc:550]: We should never be using `@ApplicationScoped` unless there is a good reason to do so. It introduces a proxy object for no good reason

**Review by gwenneg — APPROVED** [review]
LGTM

File: docs/src/main/asciidoc/cache.adoc (hunk 1 above)
Comment (gwenneg) [reply docs/src/main/asciidoc/cache.adoc:550]: Yes I know about that, but with this PR the Cache doc suggests doing the opposite of [the CDI doc](https://quarkus.io/guides/cdi#applicationscoped-and-singleton-look-very-similar-which-one-should-i-choose-for-my-quarkus-application) 😃

Comment (geoand) [reply docs/src/main/asciidoc/cache.adoc:550]: Yeah, we should fix that too :)

**Pull request merged**